components/brave_new_tab_ui/reducers/cryptoDotCom_reducer.ts

```diff
@@ -46,22 +69,40 @@ const cryptoDotComReducer: Reducer<NewTab.State | undefined> = (state: NewTab.St
       }
       break
 
-    case types.SET_ASSET_DATA:
+    case types.ALL_ASSETS_DETAILS_REQUESTED:
+      state = { ...state }
+      state.cryptoDotComState = {
+        ...state.cryptoDotComState,
+        fetchStatus: 'pending'
+      }
+      break
+
+    case types.ALL_ASSETS_DETAILS_RECEIVED:
       state = { ...state }
       state.cryptoDotComState = {
         ...state.cryptoDotComState,
+        fetchStatus: 'completed',
         charts: {
           ...state.cryptoDotComState.charts,
           ...payload.charts
         },
-        supportedPairs: reducePairs(payload.pairs)
+        supportedPairs: reducePairs(payload.pairs) || {}
```

Comment:
could `reducePairs` just return an empty object if some case is not met?

Reply:
I currently have `reducePairs` returning null if not given pairs in order to make a decision about keeping current state or new state here: https://github.com/brave/brave-core/blob/a22d1fb16f58eb61e44fd6483394de7102d5ddd9/components/brave_new_tab_ui/reducers/cryptoDotCom_reducer.ts#L115

Reply:
...but I think instead of doing that, I can change the short circuit to a ternary on line 115:

```
supportedPairs: payload.pairs ? reducePairs(payload.pairs) : state.cryptoDotComState.supportedPairs
```